src/common/containers/repository-setup.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component, PropTypes } from 'react';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+
+import { createWebhook } from '../actions/webhook';
+import { Message } from '../components/forms';
+import Spinner from '../components/spinner';
+
+import styles from './container.css';
+
+class RepositorySetup extends Component {
+  componentDidMount() {
+    const { account, repo, isPending } = this.props;
+
+    if (!isPending) {
+      this.props.dispatch(createWebhook(account, repo));
+    }
+  }
+
+  render() {
+    const { fullName, isPending, success, error } = this.props;
+
+    if (success) {
+      this.props.router.push(`/${fullName}/builds`);
```

Comment:
I guess `/setup` is kind of 'internal' route and we probably don't want to keep it in history (so users don't navigate to it directly), so I would use [`.replace`](https://github.com/ReactTraining/react-router/blob/master/docs/API.md#replacepathorloc) instead of `.push` here.